lib/src/main/java/org/esbtools/eventhandler/Promise.java

```diff
@@ -24,22 +24,24 @@
  * An asynchronous capture of potentially yet-to-come responses.
  *
  * <p>Allows code to build a {@link Future} object which will complete at some point in the future
- * with results from the {@link ResponsesHandler} passed to {@link #then(ResponsesHandler)}.
+ * with results from the {@link PromiseHandler} passed to {@link #then(PromiseHandler)}.
  *
  * @see Requester
  *
  * @param <T> The type of requests
- * @param <U> The type of responses
+ * @param <T> The type of responses
  */
-public interface ResponsePromise<T, U> {
+public interface Promise<T> extends Future<T> {
```

Comment:
Guava also has this same construct via `ListenableFuture`: https://github.com/google/guava/wiki/ListenableFutureExplained (you can block per Future .get(), as well as add callbacks).